File: game/src/main/java/com/scottlogic/hackathon/game/Player.java
package com.scottlogic.hackathon.game;

import lombok.AllArgsConstructor;
import lombok.Value;

/** A player (minion) in the game. */
@Value
@AllArgsConstructor
public class Player {
  /**
   * @return The unique id of the current player.
   */
  Id id;

  /**
   * @return The unique id of the owner of the current player.
   */
  Id owner;

  /**
   * @return The position of the current player.
   */
  Position position;

  public Player move(final Position position) {
    return new Player(id, owner, position);
  }
}

File: python-contestant/src/game/player.py
from dataclasses import dataclass, field

from game.position import Position


@dataclass(frozen=True)
class Player:
    """
    A player (minion) in the game.

    Attributes:
        id (int): The unique ID of the current player.
        owner (int): The unique ID of the owner of the current player.
        position (Position): The position of the current player.
    """
    id: int
    owner: int
    position: Position = field(compare=False)
